target/linux/ath79/image/generic.mk

```
@@ -2530,6 +2530,27 @@ define Device/ruckus_zf7025
 endef
 TARGET_DEVICES += ruckus_zf7025
 
+define Device/ruckus_gd11_common
+  $(Device/ruckus_common)
+  SOC := ar7161
+  IMAGE_SIZE := 15616k
+  BLOCKSIZE := 256k
+  DEVICE_PACKAGES := kmod-usb2 kmod-usb-chipidea2
+endef
+
+define Device/ruckus_zf7351
+  $(Device/ruckus_gd11_common)
+  DEVICE_MODEL := ZoneFlex 7351[-U]
+  DEVICE_PACKAGES += -swconfig
+endef
+TARGET_DEVICES += ruckus_zf7351
+
+define Device/ruckus_zf7363
+  $(Device/ruckus_gd11_common)
+  DEVICE_MODEL := ZoneFlex 7341/7343/7363[-U]
```

Please use the DEVICE_ALT* notation for multiple devices:
```suggestion
  DEVICE_MODEL := ZoneFlex 7363[-U]
  DEVICE_ALT0_VENDOR := Ruckus
  DEVICE_ALT0_MODEL := ZoneFlex 7341
  DEVICE_ALT1_VENDOR := Ruckus
  DEVICE_ALT1_MODEL := ZoneFlex 7343
```

Does the system automatically detect that the Wifi card is missing and some LAN ports are missing on the stripped down devices?

---

@hauke It does for missing Wi-Fi chip, doesn't for the Fast Ethernet switch, it's using fixed-link for now. Should I split this out to a separate model then, with different device tree?

---

I would prefer an extra device definition for the one with the different Ethernet switch.

---

Okay, just tested this. I'll push update shortly.